pymc3/distributions/arimastan.py

```diff
@@ -0,0 +1,289 @@
+        self.pp = pp  # number of predictors  arch
+        self.dd = dd  # number of seasonal differences
+        self.qq = qq  # number of predictions GARCH
+        self.period = period  # time series period
+        self.d1 = d1  # number of independent variables
+        self.sigma0 = sigma0  # Variance parameter
+        self.mu0 = mu0  # location parameter
+
+        # data
+
+        self.y = np.array(self.n1)  # outcome time series
+        self.yreal = np.array(self.n)  # outcome not differenced time series
+        self.xreg = np.array((self.n1, self.d1))  # matrix with independent variables
+
+        # prior data
+
+        self.prior_mu0 = np.array(4)  # prior location parameter
+        self.prior_sigma0 = np.array(4)  # prior scale parameter
+        self.prior_ar = np.array((self.p, 4))  # ar location hyper parameters
+        self.prior_ma = np.array((self.q, 4))  # ma location hyper parameters
+        self.prior_sar = np.array((self.pp, 4))  # prior arch hyper parameters
+        self.prior_sma = np.array((self.qq, 4))  # prior ma hyper parameters
+        self.prior_breg = np.array((self.d1, 4))  # prior ma hyper parameters
+        self.dinits = self.d + (self.period * self.dd)
+
+        # transformed_data
+
+        # parameters
+        self.breg = np.array(self.d1)  # regression parameters
+        self.phi0 = np.linspace(self.p, -1, 1)  # ar parameters
+        self.theta0 = np.linspace(self.q, -1, 1)  # ma parameters
+        self.pphi0 = np.linspace(self.pp, -1, 1)  # sar parameters
+        self.Theta0 = np.linspace(self.qq, -1, 1)  # sma parameters
+
+        # model_parameters
+        self.phi = np.array(self.p)  # ar parameters
+        self.theta = np.array(self.q)  # ma parameters
+        self.sphi = np.array(self.pp)  # ar parameters
+        self.stheta = np.array(self.qq)  # ma parameters
+        self.mu = np.array(self.n1)  # Mean pparameter
+        self.epsilon = np.array(self.n1)  # residual parameter
+
+    def transformation_coefficients(self):
+
+        for i in range(self.p):
+            if self.prior_ar[i, 4] == 1:
+                self.phi[i] = self.phi0[i]
+            else:
+                self.phi[i] = 2 * abs(self.phi0[i]) - 1
+
+        for i in range(self.q):
+            if self.prior_ma[i, 4] == 1:
+                self.theta[i] = self.theta0[i]
+            else:
+                self.theta[i] = 2 * abs(self.theta0[i]) - 1
+
+        for i in range(self.pp):
+            if self.prior_sar[i, 4] == 1:
+                self.sphi[i] = self.pphi0[i]
+            else:
+                self.sphi[i] = 2 * abs(self.pphi0[i]) - 1
+
+        for i in range(self.qq):
+            if self.prior_sma[i, 4] == 1:
+                self.stheta[i] = self.Theta0[i]
+            else:
+                self.stheta[i] = 2 * abs(self.Theta0[i]) - 1
+
+    # model
+
+    def prior_for_mu0(self, target, prior_for_mu0):
+        # prior mu
+        if self.prior_mu0[4] == 1:
```

Comment:
Is the idea to set prior distributions using integer arguments?

Reply:
yes pycharm pointed that out too, thinking i'll make separate utility functions